### PR TITLE
Address warnings

### PR DESCRIPTION
**Pull request by AA-Turner**
As noted in https://github.com/sphinx-doc/sphinx/issues/9777#issuecomment-1104481271, there are some new warnings from Docutils. 

This PR resolves warnings in `docutils.frontend.OptionParser`, `docutils.frontend.Option`, `docutils.nodes.Text`, and `sphinx.util.i18n.format_date`.

It switches to `image_loading` with a fallback for Docutils 0.17 and earlier, and updates a few remaining cases of `Node.traverse` -> `Node.findall`.

We also resolve all the EncodingWarnings from Sphinx directly (ones from Pytest or Docutils are not covered). 

A

### Feature or Bugfix
- Bugfix
- Refactoring

### CODE REVIEW

File: sphinx/util/docutils.py [hunk 1]
```diff
@@ -32,8 +31,6 @@
     from sphinx.config import Config
     from sphinx.environment import BuildEnvironment
 
-
-__version_info__ = version.parse(docutils.__version__).release
```
Comment (tk0miya) [inline sphinx/util/docutils.py:36]: It's better to emit a warning on accessing this attribute instead of removal. Could you revert this change, please?

File: tests/test_directive_code.py [hunk 2]
```diff
@@ -591,7 +591,7 @@ def test_linenothreshold(app, status, warning):
 def test_code_block_dedent(app, status, warning):
     app.builder.build(['dedent'])
     doctree = app.env.get_doctree('dedent')
-    codeblocks = list(doctree.traverse(nodes.literal_block))
+    codeblocks = [*doctree.findall(nodes.literal_block)]
```
Comment (tk0miya) [inline tests/test_directive_code.py]: We usually use `list()` to convert the result to a list on other code.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with nits.

**Comment by AA-Turner** [comment]
@tk0miya thanks for the review, updated.

A

**Comment by tk0miya** [comment]
Oops. conflicted. Could you update this, please? Then I'll merge this soon.

**Comment by AA-Turner** [comment]
@tk0miya merged `5.x`.

A

File: sphinx/util/docutils.py [hunk 3]
```diff
@@ -32,8 +32,14 @@
...
+deprecated_alias('sphinx.util.docutils',
+                 {
+                     '__version_info__': docutils.__version_info__,
+                 },
+                 RemovedInSphinx60Warning,
```
Comment (tk0miya) [inline sphinx/util/docutils.py:39]: We're not in hurry. So I'll remove this in 7.0. I'll change this after merging.

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Thank you for your great work! Merging.

File: sphinx/util/docutils.py (hunk 3 above)
Comment (AA-Turner) [reply sphinx/util/docutils.py:39]: Oops, I meant to change that, sorry.

A
